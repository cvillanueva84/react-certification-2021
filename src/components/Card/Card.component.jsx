import React from 'react';
import { CardWrapper, HeaderWrapper, H6, P } from './Card.styled';

function Card({ image, title, description }) {
  return (
    <CardWrapper>
      <HeaderWrapper image={image} />
      <div>
        <H6>{title}</H6>
        <P>{description}</P>
      </div>
    </CardWrapper>
  );
}

export default Card;
