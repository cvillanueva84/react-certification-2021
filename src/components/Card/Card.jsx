import React, { useContext } from 'react';
import { AppContext } from '../../providers/App/App.provider';
import {
  Container,
  ButtonCard,
  CardContent,
  TitleVideo,
  CardBody,
  ImageCard,
  LinkCard,
} from './Card.styled';

function Card(props) {
  const { image, title, description, videoId } = props;
  const { state } = useContext(AppContext);
  // console.log(videoId);
  return (
    <LinkCard to={`/view/${videoId}`}>
      <Container theme={state.theme}>
        <ButtonCard>
          <ImageCard image={image} />
        </ButtonCard>
        <CardContent>
          <TitleVideo>{title}</TitleVideo>
          <CardBody>{description}</CardBody>
        </CardContent>
      </Container>
    </LinkCard>
  );
}

export default Card;
