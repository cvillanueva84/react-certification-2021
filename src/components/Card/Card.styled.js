import styled from 'styled-components';

const CardWrapper = styled.div`
  width: 345px;
  height: 345px;
  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%),
    0px 1px 3px 0px rgb(0 0 0 / 12%);
  border-radius: 4px;
  margin: 10px;
  overflow: hidden;
`;

const HeaderWrapper = styled.div`
  background-image: ${(props) => `url(${props.image})`};
  display: block;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  height: 140px;
`;

const H6 = styled.h6`
  margin: 5px;
  font-size: 1.25rem;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 500;
  line-height: 1.6;
  letter-spacing: 0.0075em;
`;

const P = styled.p`
  font-size: 0.875rem;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 400;
  line-height: 1.43;
  color: rgba(0, 0, 0, 0.54);
`;

export { CardWrapper, HeaderWrapper, H6, P };
