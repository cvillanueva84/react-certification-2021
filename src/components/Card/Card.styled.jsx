import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Container = styled.div`
  width: 345px;
  height: 345px;
  margin: 10px;
  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%),
    0px 1px 3px 0px rgb(0 0 0 / 12%);
  border-radius: 4px;
  overflow: hidden;
  background-color: ${(props) => props.theme['background-card']};
  color: ${(props) => props.theme['text-color']};
`;

export const LinkCard = styled(Link)`
  text-decoration: none;
`;

export const ButtonCard = styled.button`
  width: 100%;
  display: block;
  text-align: inherit;
  color: inherit;
  border: 0;
  cursor: pointer;
  margin: 0;
  outline: 0;
  padding: 0;
  position: relative;
  align-items: center;
  user-select: none;
  border-radius: 0;
  justify-content: center;
  text-decoration: none;
  background-color: transparent;
`;

export const CardContent = styled.div`
  padding: 16px;
  cursor: pointer;
`;

export const TitleVideo = styled.h2`
  font-size: 1rem;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 500;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  text-align: left;
`;

export const CardBody = styled.p`
  font-size: 0.75rem;
  text-align: left;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 400;
  line-height: 1.43;
  height: 90px;
  text-overflow: ellipsis;
  overflow: hidden;
  letter-spacing: 0.01071em;
  color: rgba(0, 0, 0, 0.54);
`;
export const ImageCard = styled.div`
  display: block;
  height: 140px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-image: url(${(props) => props.image});
`;
