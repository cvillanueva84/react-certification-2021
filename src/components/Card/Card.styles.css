.card{
    border: black solid 1px;     
    width: 300px;
    height: 350px;
    margin-bottom: 2rem;
    margin-top: 2rem;
    cursor: pointer; 
}