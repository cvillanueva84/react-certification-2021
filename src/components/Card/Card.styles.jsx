import styled from 'styled-components';

const CardBody = styled.div`
  width: 100%;
  height: 20rem;
  cursor: pointer;
  overflow: hidden;
`;
const ImageCard = styled.div`
  width: 100%;
  height: 10rem;
  background-image: url(${(props) => props.imageUrl});
  background-position: center;
  background-repeat: no-repeat;
`;
const WraperText = styled.div`
  padding: 0.5rem;
`;
const Title = styled.div`
  font-weight: 600;
  font-size: 1.1rem;
`;
const Description = styled.div`
  font-size: 0.7rem;
`;

export { CardBody, ImageCard, WraperText, Title, Description };
