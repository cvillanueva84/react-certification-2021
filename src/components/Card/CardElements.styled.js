import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const CardContainer = styled.article`
  overflow: hidden;
  display: flex;
  flex-direction: column;
  background-color: var(--secondary-background-color);
  padding: 1rem;
`;

export const CardImgWrapper = styled(Link)`
  display: block;
  width: calc(100% + 2rem);
  height: 12rem;
  border-bottom: 3px solid #1c5476;
  margin: -1rem -1rem 0;
  background: var(--color-grey-1);
  cursor: pointer;
`;

export const CardImg = styled.img`
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

export const CardTitle = styled.h4`
  margin-bottom: 0%.5rem;
  font-size: 1.2rem;
  font-weight: 700;
  line-height: 1.3;
  margin-bottom: 0.625rem;
`;

export const CardDescription = styled.p`
  font-size: 0.9rem;
  color: var(--secondary-font-color);
  span {
    display: block;
    margin-top: calc(0.625rem / 2);
    color: inherit;
    font-weight: bold;
  }
`;
