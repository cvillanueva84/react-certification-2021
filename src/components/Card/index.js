export {default} from './Card.component.jsx'; 