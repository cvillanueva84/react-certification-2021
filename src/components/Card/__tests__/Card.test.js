import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';

import Card from '../index';
import data from '../../../mocks/youtube.json';
import { AppRouter } from '../../../routers/AppRouter';
import { ContextWrapper } from '../../../context/ContextWrapper';
import { ThemeProvider } from '../../../ThemeProvider';

describe('Testing Card component', () => {
  // const initialState = {
  //   search: 'wizeline',
  //   theme: true,
  //   isAuthenticated: false,
  //   auth: {},
  //   videoList: [],
  // };
  // const contextValue = {
  //   dispatch: jest.fn(),
  //   state: initialState,
  // };
  const item = data.items[1];
  let tree;

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  beforeEach(() => {
    tree = render(
      <ContextWrapper>
        <ThemeProvider>
          <HashRouter>
            <AppRouter />
            <Card item={item} path={`/favorite/${item.id}`} />
          </HashRouter>
        </ThemeProvider>
      </ContextWrapper>
    );
  });

  it('should take a snapshop', () => {
    expect(tree).toMatchSnapshot();
  });
});
