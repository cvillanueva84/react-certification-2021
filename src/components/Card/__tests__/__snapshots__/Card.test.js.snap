// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component should take a snapshop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-bdnxRM jgFpNv"
      >
        <div
          class="sc-gtsrHT hYsIYV"
          src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
        />
        <div
          class="sc-hKFxyN biVPrh"
        >
          <h2
            class="sc-dlnjwi fODMA-D"
          >
            Video Tour | Welcome to Wizeline Guadalajara
          </h2>
          <p
            class="sc-dlnjwi jnblUh"
          >
            Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
          </p>
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bdnxRM jgFpNv"
    >
      <div
        class="sc-gtsrHT hYsIYV"
        src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
      />
      <div
        class="sc-hKFxyN biVPrh"
      >
        <h2
          class="sc-dlnjwi fODMA-D"
        >
          Video Tour | Welcome to Wizeline Guadalajara
        </h2>
        <p
          class="sc-dlnjwi jnblUh"
        >
          Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
        </p>
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
